src/main/java/spoon/support/reflect/CtModifierHandler.java

```diff
@@ -57,7 +58,7 @@ public CtModifierHandler setExtendedModifiers(Set<CtExtendedModifier> extendedMo
 	}
 
 	public Set<ModifierKind> getModifiers() {
-		return modifiers.stream().map(CtExtendedModifier::getKind).collect(Collectors.toSet());
+		return modifiers.stream().map(CtExtendedModifier::getKind).collect(linkedHashSetCollector());
```

Comment:
I suggest using [`Collectors.toCollection()`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toCollection-java.util.function.Supplier-) instead of writing a collector from scratch.

```suggestion
		return modifiers.stream().map(CtExtendedModifier::getKind).collect(Collectors.toCollection(LinkedHashSet::new));
```